scapy/contrib/roce.py

```diff
@@ -228,6 +272,30 @@ class AETH(Packet):
     ]
 
 
+class DETH(Packet):
+    name = "DETH"
+    fields_desc = [
+        XIntField("qkey", 0),
+        XByteField("reserved", 0),
+        XBitField("sqp", 0, 24)
+    ]
+
+    def guess_payload_class(self, payload):
+        # type: (bytes) -> Type[Packet]
+        bth = self.underlayer
+        if isinstance(bth, BTH):
+            if bth.opcode == opcode('UD', 'SEND_ONLY_WITH_IMMEDIATE')[0]:
```

Comment:
For `opcode` to only be used here this looks kinda ridiculous tbh. a 
```
if bth.opcode == 0x05: # SEND_ONLY_WITH_IMMEDIATE
```
would have been much more sane

Reply:
For this use case, I agree, but I suppose users of this module can use `opcode` with their packets, and perhaps the module itself may use it in the future with other opcode types. I'm afraid I've only gotten to implement a part of the RoCE headers so far.

Reply:
Users can do
`BTH(opcode='SEND_ONLY_WITH_IMMEDIATE')`
since you are using a ByteEnumField

Reply:
Hey, sorry for the late response. I've uploaded a new version using a constant instead of calling through `opcode()` to build the opcode value. I hope this looks better.